Source: git commit add50e3697395a70e4d7e5a1a2748f3fc5d8718d
Upstream: yes
Reason: Without this patch, ceres-solver fails to build as a shared library

From add50e3697395a70e4d7e5a1a2748f3fc5d8718d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bj=C3=B6rn=20Piltz?= <bjornpiltz@gmail.com>
Date: Wed, 4 Jun 2014 10:23:41 +0200
Subject: [PATCH] Don't build miniglog unconditionally as a static library

This is only needed on Windows because of lacking dllexport's.

Change-Id: I7ee4b3fd640d5870af2a73d7bfa6ccc8356b2fa0
---
 internal/ceres/CMakeLists.txt | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/internal/ceres/CMakeLists.txt b/internal/ceres/CMakeLists.txt
index dd75a01..dd48ab7 100644
--- a/internal/ceres/CMakeLists.txt
+++ b/internal/ceres/CMakeLists.txt
@@ -138,7 +138,13 @@ ENDIF (SCHUR_SPECIALIZATIONS)
 # Primarily for Android, but optionally for others, use the minimal internal
 # Glog implementation.
 IF (MINIGLOG)
-  ADD_LIBRARY(miniglog STATIC miniglog/glog/logging.cc)
+  IF (MSVC)
+    # Miniglog needs to be built statically with Visual Studio since it
+    # doesn't export any symbols with dllexport/dllimport.
+    ADD_LIBRARY(miniglog STATIC miniglog/glog/logging.cc)
+  ELSE()
+    ADD_LIBRARY(miniglog miniglog/glog/logging.cc)
+  ENDIF()
   INSTALL(TARGETS miniglog
           EXPORT  CeresExport
           RUNTIME DESTINATION bin
-- 
2.0.0

